DuckDuckGo/BrowserChromeManager.swift

```diff
@@ -156,6 +157,9 @@ private class BarsAnimator {
     
     var transitionStartPosY: CGFloat = 0
     
+    private var scrollFromBottomTriggered = false
+    private var scrollFromBottomInvalidated = false
```

Comment:
Combinations of these two allow for 4 states (false false, false true, true false, true true) but I see that one combination is unused (invalid).Maybe this could be replaced with a state represented as an Enum? 

```
enum BottomTransitioning { // BottomBounceTransitioning ? BottomBounceRevealing ?
     case possible
     case triggered
     case cancelled
}
```

`possible` - equivalent to `Triggered = false ; Invalidated = false`
`triggered` - equivalent to `Triggered = true ; Invalidated = false`
`cancelled` - equivalent to `Triggered = false ; Invalidated = true`

Reply:
Agree, will make a change.